src/components/SearchModal/components/SearchInput/components/SearchInputBar/SearchInputBar.tsx

```diff
@@ -1,14 +1,19 @@
-import { API } from "@/apis/Api"
 import * as S from "./SearchInputBar.Styles"
-import { useState } from "react"
 import { Search } from "@mui/icons-material"
 import {
   SEARCH_INPUT_BAR_PLACEHOLDER,
   SEARCH_INPUT_BAR_CHANGE_TIMER_DELAY,
-} from "@/components/Search/Search.Constants"
+} from "@/components/SearchModal/SearchModal.Constants"
+import { useState, useEffect } from "react"
+import { showResultProp } from "@/components/SearchModal/SearchModal.Types"
 
-const SearchInputBar = () => {
+const SearchInputBar = ({ showResults }: showResultProp) => {
   const [searchKeyword, setSearchKeyword] = useState("")
+  const [isFocusOnBar, setIsFocusOnBar] = useState(false)
+
+  useEffect(() => {
+    showResults(searchKeyword)
+  }, [searchKeyword, showResults])
```

Comment:
[P2] showResults가 keyword를 setState로 받아서 ```const [searchKeyword, setSearchKeyword] = useState("") ```을 지우고 
useEffect도 지우고 setTimeout 내부의 setSearchKeyword 를 showResult로 바꿔도 될 것 같아요! 👍
```suggestion
  const [isFocusOnBar, setIsFocusOnBar] = useState(false)
```

Reply:
오 미쳐 발견하지 못한 부분인데 감사합니다!